src/utils/Result.ts

```diff
@@ -49,6 +49,60 @@ export class ResultPipe<T, E> {
   }
 }
 
+export type PromiseOrValue<T> = T | Promise<T>;
+
+/**
+ * Helper class for chaining together a series of `Result` operations that accepts also promises.
+ */
+export class ResultPipeAsync<T, E> {
```

Comment:
Looks like maybe you were able to get away without needing this?

Reply:
I had to resort to readFileSync, which isn't ideal. Ideally all imports should be resolved asynchronously. I didn't get to fix the actual pipeline because it was more complicated.